### DISCUSSION TITLE
How to make NavigateBackAsync take me back via different DataMap

### DISCUSSION THREAD

**Discussion by HakanL** [unoplatform/uno.extensions#1582] [Q&A]
I have routes that has a single item like this:
```
new DataViewMap<MenuPage, MenuViewModel, MenuItems>(),
```

and I navigate to a page that display menu items with this command:
```
this.navigator.NavigateViewModelAsync<MenuViewModel>(this, data: menuItems)
```
And from here I can go further down the path, with the same call, but with a different `menuItems` instance. I can navigate down the menu perfectly fine, but when I'm let's say 2 levels down and call `NavigateBackAsync` then I'm taken all the way to the top. It's like the different calls to `NavigateViewModelAsync` doesn't take the data into consideration. How can I make sure all my calls to navigate with different data are added to the navigation stack?

**Comment by HakanL** [comment]
Here's a repository that shows the issue: https://github.com/HakanL/Uno-Navigate-Issue

**Comment by HakanL** [comment]
I don't know if this is related, but my finding is that when navigating to the same view model, but with different data, a new instance of the ViewModel is created (as expected), but the Page is just re-used with a new DataContext. That may be correct, it's just an observation that perhaps is related to the navigation stack.

**Reply by nickrandolph** [reply]
Yes, your findings are correct - the current page is reused with a new viewmodel instance. An alternative to going back, would be to navigate forward with the previous data item. As you pointed out this will reuse the same page instance with a new viewmodel instance

**Reply by HakanL** [reply]
Which is fine, but when I go back I would assumed that it would have a stack of multiple ViewModels, but with the different data? Or is there a better way to track this (short of implementing my own stack)? I have a multi-level menu (top menu, then you select an item, going to the next level, same VM but with different list of menu items, and with an option to go to the parent level).
Is there a way I can catch the event when going back to the previous page? I guess it may be related to https://github.com/unoplatform/uno.extensions/discussions/1588.

**Reply by HakanL** [reply]
Is there a way to make it take the Data into consideration for the Back?
